client/src/js/base/ProgressBar.js

```diff
@@ -1,152 +1,57 @@
 import React from "react";
-import PropTypes from "prop-types";
-import CX from "classnames";
-import { round } from "lodash-es";
-import { bsStyles } from "./utils";
+import styled from "styled-components";
 
-/**
- *
- */
-export class AutoProgressBar extends React.Component {
-    constructor(props) {
-        super(props);
-        this.interval = null;
-        this.state = {
-            fill: 0,
-            active: this.props.active
-        };
-    }
-
-    static propTypes = {
-        step: PropTypes.number,
-        bsStyle: PropTypes.oneOf(bsStyles),
-        active: PropTypes.bool,
-        interval: PropTypes.number,
-        affixed: PropTypes.bool
-    };
-
-    static defaultProps = {
-        step: 30,
-        interval: 120
-    };
+const progressValueColor = {
+    success: "#60af98",
+    warning: "#be9235",
+    danger: "#af3227"
+};
 
-    static getDerivedStateFromProps(nextProps, prevState) {
-        if (prevState.active && !nextProps.active) {
-            return { fill: 100, active: nextProps.active };
-        }
+const StyledAffixedProgress = styled.progress`
```

Comment:
You can repeat yourself less by reusing styles and logic between `StyledProgress` and `StyledAffixedProgress`.

Use `StyledProgress` as a base for the affixed variant:

```js
const StyledAffixedProgress = styled(StyledProgress)``;
```